CONTRIBUTING.md

```diff
@@ -0,0 +1,435 @@
+  
+  The state of a component's Storybook page quite accurately reflects the state
+  of the component. Components with a well-designed API and no unnecessary
+  dependencies are easy to instantiate in isolation in Storybook. Conversely,
+  overly complex components are a nightmare in Storybook.
+
+- Use identified modules to import stores (using unquoted identifiers). It is
+  required for the stubs mechanism to work properly.
+
+  In practice imports like `import AppLayouts.Wallet.stores 1.0 as WalletStores`
+  should be used instead of `import "./stores" as WalletStores`. Second version is
+  relative import from the file system. As a consequence, the mechanism for
+  overriding import paths for tests and Storybook's pages will not work.
+
+## Code Style
+
+- QML code should be in-line with
+  [QML Coding Conventions](https://doc.qt.io/qt-5/qml-codingconventions.html).
+- Top level component `id` should be always `root`.
+- Private properties should be hidden in a `d` object (`QtObject { id: d }`, or
+  `QObject { id: d }`).
+- `objectName` property should not be specified for the top level component in
+  a given file.
+  
+  Component may be used in various context and usually should have different
+  name in every context to disambiguate when doing lookup in unit tests, squish
+  tests or monitoring tool.
+- `qmldir`: entries should be sorted (`Alt+Shift+S` on selection in `QtCreator`).
+- Comments and documentation should cover parts which are not obvious, can be
+  skipped where intention is clear.
+- A declarative implementation should be favoured over imperative code in QML
+  in most cases.
+- Use curly brackets to make complex expressions easier to read:
+
+  ```qml
+  readonly property bool errorMode: popup.isLoading || !recipientInputLoader.ready ?
+                                      false : errorType !== Constants.NoError
+                                      || networkSelector.errorMode
+                                      || !(amountToSendInput.inputNumberValid
+                                            || d.isCollectiblesTransfer) // Hard to read
+  ```
+
+  ```qml
+  readonly property bool errorMode: { // Easier to read
+      if (popup.isLoading || !recipientInputLoader.ready)
+          return false
+
+      return errorType !== Constants.NoError
+          || networkSelector.errorMode
+          || !(amountToSend.ready || d.isCollectiblesTransfer)
+  }
+  ```
+
+## Other
+
+- Cryptocurrency balances should be always handled as a big integer strings and
+  converted to localized human-friendly form only for displaying. Some basic rationale is
+  provided in [this ticket](https://github.com/status-im/status-desktop/issues/11376).
+
+## Useful links
+
+- [Qt Quick Layouts Overview](https://doc.qt.io/qt-6/qtquicklayouts-overview.html)
+- [Customizing Qt Quick Controls](https://doc.qt.io/qt-6/qtquickcontrols-customize.html)
+- [Keyboard Focus in Qt Quick](https://doc.qt.io/qt-6/qtquick-input-focus.html)
```

Comment:
could possibly add this as well https://www.notion.so/Strategy-for-Numbers-and-String-in-Swap-and-Generally-5fed06457c134c82ab893e2b0da03c6e

Reply:
Will add by a chance!